src/main/java/org/radarcns/management/repository/filters/UserFilter.java

```diff
@@ -47,8 +47,13 @@ public Predicate toPredicate(Root<User> root, CriteriaQuery<?> query, CriteriaBu
             predicates.add(cb.like(cb.lower(authorityJoin.get("name")),
                     "%" + authority.trim().toLowerCase() + "%"));
         }
-        return predicates.isEmpty() ? null
-                : cb.and(predicates.toArray(new Predicate[predicates.size()]));
+
+        if (predicates.isEmpty()) {
+            return null;
+        } else {
+            query.distinct(true);
+            return cb.and(predicates.toArray(new Predicate[predicates.size()]));
```

Comment:
For future reference: `predicates.toArray(new Predicate[0])` is faster to execute and shorter than `predicates.toArray(new Predicate[predicates.size()])`

Reply:
Apparently PMD 1.8 advises other way around. 
Calls to a collection's toArray() method should specify target arrays sized to match the size of the collection. Initial arrays that are too small are discarded in favour of new ones that have to be created that are the proper size.
```
List foos = getFoos();

    // inefficient, the array will be discarded
Foo[] fooArray = foos.toArray(new Foo[0]);

    // much better; this one sizes the destination array,
    // avoiding of a new one via reflection
Foo[] fooArray = foos.toArray(new Foo[foos.size()]);
```

Reply:
It used to be like that when PMD 1.8 came out, but since Java 5 the initialiser with 0 is faster. For a full review: https://shipilev.net/blog/2016/arrays-wisdom-ancients/ ([Conclusion](https://shipilev.net/blog/2016/arrays-wisdom-ancients/#_conclusion))

Reply:
Yes. I read about it. I will also look into upgrading PMD version

Reply:
Probably the rule is still in there, you can disable it in the performance ruleset, by adding
```xml
<exclude rule="OptimizableToArrayCall"/>
```